src/DynamoCore/Engine/ShortestQualifiedNameReplacer.cs

```diff
@@ -106,7 +106,7 @@ public override AssociativeNode VisitIdentifierNode(IdentifierNode node)
             // return IdentifierNode if identifier string is not separated by '.'
             if (strIdentList.Length == 1)
             {
-                return new IdentifierNode(strIdentList[0]);
+                return node;
```

Comment:
If my memory serves correctly, then the AST walkers are a bit inconsistent when it comes to returning new AST nodes or the same node - maybe this should return a copy of the node instead?

Reply:
The AST walkers are meant to return new nodes _only_ if they are rewritten and return the original node if there is no change.

Reply:
ah! Is that documented somewhere? Would it make sense to add a note about that to the base class or something? Not that it needs to be part of this pr.

Reply:
Can do.